Library/OcBootManagementLib/BootAudio.c

```diff
@@ -178,50 +175,65 @@ OcPlayAudioEntry (
   IN     OC_BOOT_ENTRY      *Entry
   )
 {
-  OcPlayAudioFile (Context, OcVoiceOverAudioFileIndexBase + Entry->EntryIndex, FALSE);
+  EFI_STATUS  Status;
+  CHAR8       BasePath[8];
+
+  if ((Entry->EntryIndex > 0) && (Entry->EntryIndex <= 9 + 26)) {
+    Status = OcAsciiSafeSPrint (
+               BasePath,
+               sizeof (BasePath),
+               "%a%c",
+               Entry->EntryIndex > 9 ? "Letter" : "",
+               "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ"[Entry->EntryIndex]
+               );
+    ASSERT_EFI_ERROR (Status);
```

Comment:
Both assert and if sound weird. Feels like it should be not checked at all or asserted as all the data is predictable and trusted.

Reply:
Hope the revised version with static calculation of required string size looks okay? (Should maybe, defensively, still have ASSERT?)